### ISSUE TITLE
`tag_s!` panics when parsing Unicode `&str`

### ISSUE THREAD

**Issue by AndyShiue** [Geal/nom#369]
```rust
#[macro_use]
extern crate nom;

fn main() {
    named!(dot(&str) -> &str,
        tag_s!(".")
    );
    dot("點");
}
```

The above code yields the following error:

```
Yung-Hueis-MBP:osz_parse andyshiue$ RUST_BACKTRACE=1 cargo run
   Compiling osz_parse v0.0.0 (file:///Users/andyshiue/Desktop/osz_parse)
    Finished debug [unoptimized + debuginfo] target(s) in 0.32 secs
     Running `target/debug/osz_parse`
thread 'main' panicked at 'index 0 and/or 1 in `點` do not lie on character boundary', ../src/libcore/str/mod.rs:1721
stack backtrace:
   1:        0x109eec3d8 - std::sys::backtrace::tracing::imp::write::h6f1d53a70916b90d
   2:        0x109eede8f - std::panicking::default_hook::{{closure}}::h137e876f7d3b5850
   3:        0x109eed425 - std::panicking::default_hook::h0ac3811ec7cee78c
   4:        0x109eed976 - std::panicking::rust_panic_with_hook::hc303199e04562edf
   5:        0x109eed814 - std::panicking::begin_panic::h6ed03353807cf54d
   6:        0x109eed732 - std::panicking::begin_panic_fmt::hc321cece241bb2f5
   7:        0x109eed697 - rust_begin_unwind
   8:        0x109f12b90 - core::panicking::panic_fmt::h27224b181f9f037f
   9:        0x109f130c0 - core::str::slice_error_fail::h3b64334b3fabf062
  10:        0x109ee7554 - core::str::traits::<impl core::ops::Index<core::ops::RangeTo<usize>> for str>::index::h93d6b3a34b82fcd2
  11:        0x109ee7a4e - osz_parse::main::dot::hff0fd767b1853439
  12:        0x109ee795e - osz_parse::main::h9811fbed03ff9e54
  13:        0x109eee44a - __rust_maybe_catch_panic
  14:        0x109eecf66 - std::rt::lang_start::h538f8960e7644c80
  15:        0x109ee7cd9 - main
```

**Comment by AndyShiue** [comment]
Oh wait, In the source code `tag_s!` is just a thin wrapper around `tag!()` lol

**Comment by AndyShiue** [comment]
https://github.com/Geal/nom/blob/ac8fe712b9f8b3da661828ffc5b97a825007b590/tests/blockbuf-arithmetic.rs#L120

I guess the problem is here.

**Comment by Geal** [comment]
Fixed in 28414ed. Iterating over the chars feels better than doing a `slice_unchecked`
